src/network/network_gui.cpp

```diff
@@ -150,15 +152,21 @@ class NWidgetServerListHeader : public NWidgetContainer {
 		}
 
 		/* All remaining space goes to the first (name) widget */
-		this->head->current_x = given_width;
+		this->children.front()->current_x = given_width;
 
 		/* Now assign the widgets to their rightful place */
 		uint position = 0; // Place to put next child relative to origin of the container.
-		for (NWidgetBase *child_wid = rtl ? this->tail : this->head; child_wid != nullptr; child_wid = rtl ? child_wid->prev : child_wid->next) {
+		auto assign_position = [&](const std::unique_ptr<NWidgetBase> &child_wid) {
 			if (child_wid->current_x != 0) {
 				child_wid->AssignSizePosition(sizing, x + position, y, child_wid->current_x, this->current_y, rtl);
 				position += child_wid->current_x;
 			}
+		};
+
+		if (rtl) {
+			std::for_each(std::rbegin(this->children), std::rend(this->children), assign_position);
+		} else {
+			std::for_each(std::begin(this->children), std::end(this->children), assign_position);
 		}
```

Comment:
Not a huge fan of this. I suppose
```
if (rtl) {
    start = rbegin...
    end = rend...
} else {
    start = begin...
    end = end...
}
for (auto it = start; it != end; ++it) {
    ...
}
```
won't work due to type differences?

Reply:
Indeed, sadly forward and reverse iterators can be mixed like that, it would basically need two separate duplicate loops. And this way basically is, but done behind for_each.

Reply:
Might it make sense to reverse the order of `this->children` when it's a horizontal container widget and rtl is active? Though that would probably be way out of scope for this PR.

Reply:
If I reverse the widgets then the loop that determines which header should be hidden needs to be reversed depending on RTL. Can't win!